src/transformers/generation_utils.py

```diff
@@ -1309,6 +1311,17 @@ def generate(
                 "Diverse beam search cannot be used in sampling mode. Make sure that `do_sample` is set to `False`."
             )
 
+        if typical_p is not None:
+            if typical_p < 0 or typical_p > 1:
+                raise ValueError("Value of typical_p must be between 0 and 1.")
+            elif do_sample is False:
+                raise ValueError(
+                    "Decoder argument `typical_p` must be used in sampling mode. "
+                    "Make sure that `do_sample` is set to `True`."
+                )
```

Comment:
This is a problem for several other options, such as `top_k` or `length_penalty`, which have no effect in some generation modes.

Despite it being a correct check, we are assessing how to best handle it. We would like to avoid having an if/else branch for each input variable, as it would make `generate` even longer and less readable. Since it is harmless to have `typical_p` set with `do_sample=False` and we don't want to promote further checks of this kind, I'm going to ask to remove it for now.

Reply:
OK, I do hope that the `do_sample` check can happen in the future because receiving `typical_p` but doing greedy or non-sampled beam decoding is, in my view, silently failing.

Removed this and committed requested changes ✅